linkerd/proxy/api-resolve/src/metadata.rs

```diff
@@ -31,6 +31,9 @@ pub enum ProtocolHint {
     Unknown,
     /// The destination can receive HTTP2 messages.
     Http2,
+    /// The destination will handle all connections as opaque TCP, so do not
+    /// send a session protocol in a tagged transport connection.
```

Comment:
This is a downstream implication of this hint 

```suggestion
    /// The destination will handle traffic as opaque, regardless of
    /// the local proxy's handling of the traffic.
```

The details of tagged transport are out of scope here. We should be clear about what the setting means, not enumerate all of the behaviors that might rely on this setting.